x-pack/legacy/plugins/apm/public/components/app/TransactionDetails/WaterfallWithSummmary/index.tsx

```diff
@@ -101,20 +103,45 @@ function MaybeViewTraceLink({
 }
 
 interface Props {
-  transaction: ITransaction;
   urlParams: IUrlParams;
   location: Location;
   waterfall: IWaterfall;
   exceedsMax: boolean;
+  isLoading: boolean;
 }
 
-export const Transaction: React.SFC<Props> = ({
-  transaction,
+export const WaterfallWithSummmary: React.SFC<Props> = ({
   urlParams,
   location,
   waterfall,
-  exceedsMax
+  exceedsMax,
+  isLoading
 }) => {
+  const { entryTransaction } = waterfall;
+  if (!entryTransaction) {
+    if (isLoading) {
+      return (
```

Comment:
Maybe we can do this without multiply nested return statements:

```ts
if (!entryTransaction) {
    const content = isLoading ? (
      <LoadingStatePrompt />
    ) : (
      <EuiEmptyPrompt
        title={
          <div>
            {i18n.translate('xpack.apm.transactionDetails.traceNotFound', {
              defaultMessage: 'The selected trace cannot be found'
            })}
          </div>
        }
        titleSize="s"
      />
    );

    return <EuiPanel paddingSize="m">{content}</EuiPanel>;
  }
```

Reply:
Will fix